### DISCUSSION TITLE
change cache for nova.version

### DISCUSSION THREAD

**Discussion by niekdemelker** [laravel/nova-issues#6288] [Ideas & Feature Requests] [answered]
In telescope is noticed all these cache misses for 'nova.version'. After some searching i found that the version is being pulled directly from the composer file, because of this there is the next bit of code surrounding it.

```php
Cache::driver('array')->rememberForever('nova.version', function () {
    //
}
```

 The result is every pageload misses and hits the cache (mostly one miss and one hit per request, and nova triggers a lot of requests).

I like to propose a change. Because Nova requires Spatie\Once, maybe this can replace the cache, and also instead of using the composer.json to read the version, maybe this can be replaced with `\Composer\InstalledVersions::getPrettyVersion('laravel/nova')`, just like Laravel Pulse is doing this (don´t know if this has a performance hit).

**Comment by crynobone** [comment, marked as answer]
Once v2 has a bug with caching scalar values such as `string`. Which is the real reason we moved to `Cache`. This will be done once we remove support for PHP 7.x for Nova and remove support for Once v1 and v2.

**Reply by niekdemelker** [reply]
Thank you for the response. I will just add the cache key to all the Telescope and Pulse ignore lists and wait till nova requires Laravel 11 ;)

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
